models/__init__.py

```diff
@@ -0,0 +1,279 @@
+        '''
+        Get all minutes for a skater, by uSkaterUUID.
+        '''
+
+        with Session() as s:
+            minutes = (
+                s.query(func.sum(Ice_Time.ice_time))
+                .where(Ice_Time.uSkaterUUID == self.uSkaterUUID)
+                .scalar()
+                        )
+        return minutes
+
+    def ice_time_config_in_minutes(self, uSkaterConfig):
+        '''
+        Get all minutes for a specific configuration of a skater.
+        Requires the uSkaterConfig as an ID to match on.
+        '''
+
+        with Session() as s:
+            minutes = (
+                s.query(func.sum(Ice_Time.ice_time))
+                .where(Ice_Time.uSkaterUUID == self.uSkaterUUID)
+                .where(Ice_Time.uSkaterConfig == uSkaterConfig)
+                .scalar()
+                        )
+        return minutes
+
+
+class insert_data():
+
+    def add_bulk_session(ice_session):
+        '''
+        This should probably be moved to ice_sessions class
+        '''
+
+        with Session() as s:
+            for session in ice_session:
+                s.add(Ice_Time(**session))
+                s.commit()
+        s.close()
+
+
+class ice_sessions():
+
+    def list_all(uSkaterUUID):
+        '''
+        lists all ice sessions of a particular skater via uSkaterUUID
+        '''
+
+        with Session() as s:
+            all_sessions = (
+                s.query(Ice_Time, Locations, IceType)
+                .where(Ice_Time.uSkaterUUID == uSkaterUUID)
+                .join(Locations, Ice_Time.rink_id == Locations.rink_id)
+                .join(IceType, Ice_Time.skate_type == IceType.id)
+                .all()
+                )
+
+        return all_sessions
+
+    def df_all(uSkaterUUID):
+        '''
+        Same as list_all(), but returns a pandas dataframe which
+        may be the way of the future if we can pass json correctly
+        Returns same data as legacy sessions table:
+        date, ice session meta, coach meta, rink meta.
+        '''
+
+        df = pd.read_sql_query(
+            sql=Session().query(
+                Ice_Time.date,
+                Ice_Time.ice_time,
+                Ice_Time.ice_cost,
+                IceType.ice_type,
+                Ice_Time.coach_time,
+                Coaches.coach_Fname,
+                Coaches.coach_Lname,
+                Ice_Time.coach_cost,
+                Locations.rink_name,
+                Locations.rink_city,
+                Locations.rink_state,
+                )
+            .where(Ice_Time.uSkaterUUID == uSkaterUUID)
+            .join(Locations, Ice_Time.rink_id == Locations.rink_id)
+            .join(IceType, Ice_Time.skate_type == IceType.id)
+            .join(Coaches, Ice_Time.coach_id == Coaches.id)
+            .statement, con=engine
+        )
+
+        return df
+
+
+class equipment():
+    '''
+    Everything related to equipment ...
+    '''
+
+    class add():
+
+        def boot(boots):
```

Comment:
I have to admit, I'm not sure about this, and in a few ways.  Ive been able to use this code to insert rows as expected..... so `self` being the first arg doesnt seem to affect anything...  How ever, I'm not sure how/why that makes an impact here because Im lacking the understanding behind why this is the behavior.  

Ultimately, i want to structure things in a way that i can import models, which may actually be the wrong verbiage now that i think about it, and do something like...

```
from models import equipment as equip

equip.add.boots(some_dict) # add some new boots
equip.table.boots(uSkaterUUID) # dump all rows of previous boots
```

Perhaps there is a better way of doing that instead of using classes to structure this??

Reply:
I do reference the organization/structure a little bit more in #5